### PR TITLE
File security client basic structure with create certificate API

### PR DESCRIPTION
**Pull request by smahapatra86**
- Created skeleton for client project
- Added swagger v2.0 and autorest script to generate code
- Added custom classes to use the autogenerated client class
- Added CreateCertificate method in the custom client class to create a certificate through autogenerated class
- Added skeleton for sample project which is going to use the client class

### CODE REVIEW

**Review by avin101 — APPROVED** [review]

File: src/Kmd.Logic.FileSecurity.Client/filesecurity.swagger.json [hunk 1]
```diff
@@ -0,0 +1,201 @@
+{
+  "swagger": "2.0",
```
Comment (avin101) [inline src/Kmd.Logic.FileSecurity.Client/filesecurity.swagger.json:2]: Why not use latest openapi spec i.e. 3.1 here?

Comment (smahapatra86) [reply src/Kmd.Logic.FileSecurity.Client/filesecurity.swagger.json:2]: To use openapi 3.0.1 we need to use the latest autorest version as the autorest version (2.0 or legacy version) which we are using will not support openapi 3.0.1. Using autorest latest version to generate code gave some issues and it is different the way code is generated using legacy version. Few e.g. 
1. It was creating a separate csproj file outside the current project folder.
2. It was throwing few build errors which is not the case when legacy version is used.
3. It was not generating an client interface. It was generating a class directly.
We need to analyze these issues before going for latest autorest and openapi version. Hence in order to proceed with current client creation we went for openapi v2.
We can try to find the right way to use latest versions and then update later. :)

File: src/Kmd.Logic.FileSecurity.Client/FileSecurityOptions.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    /// </summary>
+    public sealed class FileSecurityOptions
+    {
+        /// <summary>
+        /// Gets or sets the Logic FileSecurity service.
+        /// </summary>
+        /// <remarks>
+        /// This option should not be overridden except for testing purposes.
+        /// </remarks>
+        public Uri FileSecurityServiceUri { get; set; } = new Uri("https://gateway.kmdlogic.io/file-security/v2");
```
Comment (thecorrado) [inline src/Kmd.Logic.FileSecurity.Client/FileSecurityOptions.cs]: v2?

File: src/Kmd.Logic.FileSecurity.Client/FileSecurityClient.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    public sealed class FileSecurityClient : LogicHttpClientProvider
+    {
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileSecurityClient"/> class.
+        /// </summary>
+        /// <param name="httpClient">The HTTP client to use. The caller is expected to manage this resource and it will not be disposed.</param>
+        /// <param name="tokenProviderFactory">The Logic access token provider factory.</param>
+        /// <param name="options">The required configuration options.</param>
+        public FileSecurityClient(HttpClient httpClient, LogicTokenProviderFactory tokenProviderFactory, FileSecurityOptions options)
```
Comment (thecorrado) [inline src/Kmd.Logic.FileSecurity.Client/FileSecurityClient.cs]: I would use `ITokenProviderFactory` abstraction. See this PR https://github.com/kmdlogic/kmd-logic-cpr-client/commit/42107d6d5f72671be6f2ab26dfa80fef45677475#diff-1a390da90b6b76ed9245ecc74bcc270b451723b95d3a031ec03d6d4528c94f1f

File: src/Kmd.Logic.FileSecurity.Client/LogicHttpClientProvider.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
+﻿using Kmd.Logic.FileSecurity.Client.Types;
+using Kmd.Logic.Identity.Authorization;
+using Microsoft.Rest;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+
+namespace Kmd.Logic.FileSecurity.Client
+{
+    public class LogicHttpClientProvider
```
Comment (thecorrado) [inline src/Kmd.Logic.FileSecurity.Client/LogicHttpClientProvider.cs]: Not needed. We can use `LogicTokenProviderFactory` from https://github.com/kmdlogic/kmd-logic-identity-authorization

File: src/Kmd.Logic.FileSecurity.Client/ModelMarshaller.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
+﻿using Kmd.Logic.FileSecurity.Client.Models;
+using Kmd.Logic.FileSecurity.Client.ServiceMessages;
+using Kmd.Logic.FileSecurity.Client.Types;
+using Microsoft.Rest;
+using System.Threading.Tasks;
+
+namespace Kmd.Logic.FileSecurity.Client
+{
+    internal static class ModelMarshaller
```
Comment (thecorrado) [inline src/Kmd.Logic.FileSecurity.Client/ModelMarshaller.cs]: As for now I don't see any value of having that. In docgen module there was a reason to have that, here not really. Let's use models from generated client directly

**Review by thecorrado — CHANGES_REQUESTED** [review]

File: src/Kmd.Logic.FileSecurity.Client/FileSecurityOptions.cs (hunk 2 above)
Comment (smahapatra86) [reply src/Kmd.Logic.FileSecurity.Client/FileSecurityOptions.cs]: changed to v1

File: src/Kmd.Logic.FileSecurity.Client/LogicHttpClientProvider.cs (hunk 4 above)
Comment (smahapatra86) [reply src/Kmd.Logic.FileSecurity.Client/LogicHttpClientProvider.cs]: Removed the class.

File: src/Kmd.Logic.FileSecurity.Client/ModelMarshaller.cs (hunk 5 above)
Comment (smahapatra86) [reply src/Kmd.Logic.FileSecurity.Client/ModelMarshaller.cs]: Removed the class.

File: src/Kmd.Logic.FileSecurity.Client/FileSecurityClient.cs (hunk 3 above)
Comment (smahapatra86) [reply src/Kmd.Logic.FileSecurity.Client/FileSecurityClient.cs]: Used `ITokenProviderFactory` as suggested.

**Review by thecorrado — APPROVED** [review]

**Pull request merged**